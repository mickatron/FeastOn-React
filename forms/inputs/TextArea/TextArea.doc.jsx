/* eslint-disable */
import React from 'react';
import Component from './TextArea';
import PropTypes from 'prop-types';

/**
 * A TextArea.
 *
 * 
 * @composedWith UiWrapper
*/
function TextArea(props) {
  return <Component {...props} />;
}

export default TextArea;
