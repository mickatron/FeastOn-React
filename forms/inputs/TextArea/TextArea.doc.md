### Default TextArea.
```js
<TextArea 
  value="Some text area text."
  onChange={()=>{}}
/>
```