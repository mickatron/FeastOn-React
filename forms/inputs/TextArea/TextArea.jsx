import uiWrapper from '../../../HOC/uiWrapper/uiWrapper';

// return HOC wrapped component
export default uiWrapper({
  name: 'TextArea',
  element: 'textarea',
  modifiers: [
    'small',
    'medium',
    'large',
  ],
});
