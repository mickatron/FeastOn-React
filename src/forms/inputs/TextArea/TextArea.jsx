import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from 'js/shared/proptypes/className';


export default class TextArea extends React.PureComponent {
  
  static propTypes = {
    className: classNameType,
    children: PropTypes.node.isRequired,
    rows: PropTypes.number,
    size: PropTypes.oneOf([
      'small',
      'medium',
      'large',
    ]),
  };

  static defaultProps = {
    size: 'medium',
    rows: 5,
  };

  render() {
    const {
      className,
      children,
      size,
      ...other
    } = this.props;

    const otherClassNames = {
      ['c-TextArea--' + size]: size !== 'medium',
    };

    return (
      <textarea
        className={classnames(className, 'c-TextArea', otherClassNames)}
        type="text"
        { ...other }
      >
        {children}
      </textarea>
    );
  }
}
