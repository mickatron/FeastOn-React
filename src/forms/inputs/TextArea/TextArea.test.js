
import { shallow } from 'enzyme';
import React from 'react';
import TextArea from './TextArea';

describe('TextArea', () => {

  let component = null;
  it('renders',() => {
    component = shallow(
      <TextArea
        className="test"
        rows={5}
        columns={10}
      >
        Some default value text
      </TextArea>
    );
    expect(component).toMatchSnapshot();
  });

});