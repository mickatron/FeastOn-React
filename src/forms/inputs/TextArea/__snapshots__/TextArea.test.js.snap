// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders 1`] = `
<textarea
  className="test c-TextArea"
  columns={10}
  rows={5}
  type="text"
>
  Some default value text
</textarea>
`;
